### PR TITLE
add utf8 support for TextField widget

### PR DESCRIPTION
**Pull request by zilongshanren**
1. refactor utf8_substr method to ui::Helper class
2. add utf8 support for TextField

### CODE REVIEW

**Comment by minggo** [comment]
[ci rebuild]

**Comment by minggo** [comment]
Please fix compiling error.
It seems that `unsigned long` is not supported in lua binding.

**Comment by zilongshanren** [comment]
@minggo
ok

**Comment by zilongshanren** [comment]
[ci rebuild]

**Comment by zilongshanren** [comment]
[ci rebuild]

**Comment by zilongshanren** [comment]
[ci rebuild]

File: cocos/ui/UIHelper.h [hunk 1]
```diff
@@ -66,6 +66,18 @@ class CC_GUI_DLL Helper
...
     static Widget* seekActionWidgetByActionTag(Widget* root, int tag);
+    
+    /**
+     * @brief Get a utf8 substring from a std::string with a given start position and length
+     *  Sample:  std::string str = "中国中国中国”;  substr = utf8_substr(str,0,2) will = "中国"
+     * @param start The start position of the substring.
+     * @param length The length of the substring in utf8 count
+     * @return a utf8 substring
+     */
+    static std::string utf8_substr(const std::string& str,
```
Comment (minggo) [inline cocos/ui/UIHelper.h]: Don't use `_` to connect words. It doesn't obey cocos2d-x c++ coding style.

Comment (zilongshanren) [inline cocos/ui/UIHelper.h]: ok,I will change it

**Comment by zilongshanren** [comment]
[ci rebuild]

**Comment by zilongshanren** [comment]
[ci rebuild]

**Comment by zilongshanren** [comment]
[ci rebuild]

**Comment by minggo** [comment]
[ci rebuild]

**Comment by zilongshanren** [comment]
[ci rebuild]

**Pull request merged**